src/Confluent.Kafka/Internal/Extensions/String.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2016-2017 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Text;
+
+
+namespace Confluent.Kafka
+{
+    /// <summary>
+    ///     Extension methods for the <see cref="String"/> class.
+    /// </summary>
+    internal static class StringExtensions
+    {
+        internal static Encoding ToEncoding(this string encodingName) 
+        {
+            switch (encodingName)
+            {
+                case "UTF8":
+                    return Encoding.UTF8;
+                case "UTF7":
+                    return Encoding.UTF7;
+                case "UTF32":
+                    return Encoding.UTF32;
+                case "ASCII":
+                    return Encoding.ASCII;
+                case "Unicode":
+                    return Encoding.Unicode;
+                case "BigEndianUnicode":
+                    return Encoding.BigEndianUnicode;
+                default:
+                    throw new ArgumentException("unknown string encoding: " + encodingName);
```

Comment:
Yep, this is better.

In general, I like the philosophy of there only being one way to do things, but I'm thinking I'll leave the switch statement here because the cases match up exactly with inline C# configuration (which people are likely to try) and I don't think having more than one variant matters. However, I'll point to the below URL for the available options in the API docs:

https://msdn.microsoft.com/en-us/library/system.text.encoding(v=vs.110).aspx